tests/gcp/installer/util/functional_test.py

```diff
@@ -0,0 +1,307 @@
+
+TEST_RESOURCE_DIR_PATH = os.path.join(
+    os.path.dirname(__file__), 'test_data')
+
+
+class FunctionalTest(ForsetiTestCase):
+
+    def deep_sort(self, obj):
+        """Recursively nested lists or dictionaries.
+
+        Args:
+            obj (object): Objects to sort.
+        """
+
+        if isinstance(obj, dict):
+            _sorted = {}
+            for key in sorted(obj):
+                _sorted[key] = self.deep_sort(obj[key])
+
+        elif isinstance(obj, list):
+            new_list = []
+            for val in obj:
+                new_list.append(self.deep_sort(val))
+            _sorted = sorted(new_list)
+
+        else:
+            _sorted = obj
+
+        return _sorted
+
+    def test_merge_conf_yaml_files_1_1_10(self):
+        """Merge 2 conf yaml files."""
+        self.maxDiff = None
+        # Load yaml files
+        merge_to_path = os.path.join(TEST_RESOURCE_DIR_PATH,
+                                 'merge_to_forseti_conf_server.yaml')
+        merge_from_path = os.path.join(TEST_RESOURCE_DIR_PATH,
+                                   'merge_from_forseti_conf_1.1.10.yaml')
+        merged_path = os.path.join(TEST_RESOURCE_DIR_PATH,
+                                   'expected_merged_forseti_conf_server_from_1.1.10.yaml')
+
+        merge_to = files.read_yaml_file_from_local(merge_to_path)
+        merge_from = files.read_yaml_file_from_local(merge_from_path)
+        expected_merged_obj = files.read_yaml_file_from_local(merged_path)
+
+        # Merge target into base, ignore gcs_path and output_path
+        fields_to_ignore = [
+            'db_host', 'db_user', 'db_name',
+            'inventory', 'output_path', 'gcs_path',
+            'groups_service_account_key_file',
+            'domain_super_admin_email',
+            'max_admin_api_calls_per_100_seconds',
+            'max_appengine_api_calls_per_second',
+            'max_bigquery_api_calls_per_100_seconds',
+            'max_cloudbilling_api_calls_per_60_seconds',
+            'max_compute_api_calls_per_second',
+            'max_container_api_calls_per_100_seconds',
+            'max_crm_api_calls_per_100_seconds',
+            'max_iam_api_calls_per_second',
+            'max_servicemanagement_api_calls_per_100_seconds',
+            'max_sqladmin_api_calls_per_100_seconds',
+            'max_results_admin_api']
+        field_identifiers = {'scanners': 'name',
+                             'resources': 'resource',
+                             'pipelines': 'name'}
+
+        merge_engine.merge_object(merge_from, merge_to,
+                                  fields_to_ignore, field_identifiers)
+
+        forseti_server_installer.ForsetiServerInstaller._swap_config_fields(
+            merge_from, merge_to)
+
+        merge_to = self.deep_sort(merge_to)
+        expected_merged_obj = self.deep_sort(expected_merged_obj)
+
+        self.assertEqual(expected_merged_obj, merge_to)
+
+    def test_merge_conf_yaml_files_1_1_9(self):
```

Comment:
This naming is so awesome & clear, it's a pleasure to read.

Reply:
Yay!